WT-7616 Create a pass or fail test based on many collection workgen workload

etienneptl opened the pull request:
Creation of a script for workgen perf assessment.

etienneptl commented on bench/workgen/runner/workgen_perf_check.sh:
```diff
@@ -0,0 +1,37 @@
+fi
+
+# Check if one DROP operation took too long
+THRESHOLD=45
```
The latencies have largely exceeded this value in the recent build...
http://build.wiredtiger.com:8080/job/WT-7616-plots/plot/

etienneptl replied:
As discussed with @sulabhM, instead of hardcoding thresholds, we could get them from arguments.
For now, we need more runs of the Jenkins jobs to define them.

raviprakashgiri29 commented on bench/workgen/runner/many-dhandle-stress.py:
```diff
@@ -90,29 +90,35 @@
-thread0 = Thread(ops)
-thread0.options.throttle=1000
-thread0.options.throttle_burst=1.0
+insert_thread = Thread(ops)
```
This workgen workload was generated from `wtperf/runners/many-dhandle-stress.wtperf`, So I would suggest not changing the thread names. It is understood from the operation (Operation.OP_SEARCH/Operation.OP_INSERT) parameter about the threads.

etienneptl replied:
Understood, will revert.

raviprakashgiri29 replied:
Thanks @etienneptl

raviprakashgiri29 commented on bench/workgen/runner/workgen_perf_check.sh:
```diff
@@ -0,0 +1,37 @@
+#! /bin/bash
+
```
Who calls this script? Can you please put a comment here stating the usage of this script?

etienneptl replied:
This script will be used in an evergreen task, see WT-7679.
I will add a comment about it.

sulabhM commented on bench/workgen/runner/workgen_perf_check.sh:
```diff
@@ -0,0 +1,69 @@
+THRESHOLD=$3
+VALUE=$(grep -i "cycling idle.*drop" $OUTPUT | awk '{print $9}' | sort -n | tail -1)
+if [[ $VALUE -ge $THRESHOLD ]]; then
+    echo "ERROR: One drop operation took too long: ${VALUE}s (max: ${THRESHOLD}s)"
```
Instead of "One drop operation took ..." can we say "A drop operation took.."
Also similar to above comment, instead of "max:" we can say "max allowed:"

sulabhM commented on bench/workgen/runner/workgen_perf_check.sh:
```diff
@@ -0,0 +1,69 @@
+THRESHOLD=$2
+VALUE=$(grep -ic "cycling idle.*drop" $OUTPUT)
+if [[ $VALUE -ge $THRESHOLD ]]; then
+    echo "ERROR: Too many long DROP operations: $VALUE (max: $THRESHOLD)"
```
can we say "max allowed:" instead of "max:"

sulabhM commented on bench/workgen/runner/workgen_perf_check.sh:
```diff
@@ -0,0 +1,69 @@
+THRESHOLD=$4
+VALUE=$(grep -ic "max latency exceeded" $OUTPUT)
+if [[ $VALUE -ge $THRESHOLD ]]; then
+    echo "ERROR: Too many long read/insert/update operations: $VALUE (max: $THRESHOLD)"
```
Please change "max:" to "max allowed:"

sulabhM reviewed (approved):
I have some minor comments, other than that LGTM

raviprakashgiri29 reviewed (approved):
LGTM

etienneptl commented:
Requesting your review again as I reworked the script.
Added an input argument to specify the output filename and added traces. See below how it's used:
```
$ ./workgen_perf_check.sh  many-dhandle-stress.py 1 1 1
Current Time : 2021.06.21-02.10.53
New FileName:  output_2021.06.21-02.10.53.log
$ ./workgen_perf_check.sh many-dhandle-stress.py output.log 1 1 1
python3 many-dhandle-stress.py 2>&1 | tee output.log
WARNING: Cycling idle table failed because DROP took 3 s which is longer than configured acceptable maximum of 2 s. Diff is 1 s.
WARNING: Cycling idle table failed because DROP took 3 s which is longer than configured acceptable maximum of 2 s. Diff is 1 s.
Number of long drop operations: 2
Number of long read/insert/update operations: 0
Five worst drop operations (s): 3 3 
Too many long DROP operations: 2 (max allowed: 1)
A drop operation took too long: 3s (max allowed: 1s)
FAILED
```

raviprakashgiri29 reviewed (approved):
LGTM
